tools/pylib/boututils/datafile.py

```diff
@@ -628,6 +629,12 @@ def find_dim(dim):
             # Get new list of variables, and turn into a tuple
             dims = tuple(map(find_dim, dlist))
 
+            if name in self.handle.variables:
+                raise IOError("Variable %s already present",name)
```

Comment:
This should be `ValueError` -- `IOError` means "something went wrong at the disk level"